source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1177,13 +1177,18 @@ updateRSFromPrimary
     if topologyDescription.setName is null:
         topologyDescription.setName = description.setName
 
-    else if topologyDescription.setName != description.setName:
+    if topologyDescription.setName != description.setName:
```

Comment:
Ah, good catch, though orthogonal to the rest of the PR.  Is it true that the spec tests already assume that there is no "else" here?

Reply:
As far as I can tell omitting "else" does not change the behavior in any way and just makes the case where topologydescription.setname was null less efficient.